File: src/Application Specific/Classes/Entity Collections/Spares/Product/ProductAssociatedPart/ProductAssociatedPartSQL.cs
using System.Data;

namespace FSM.Entity
{
    namespace ProductAssociatedParts
    {
        public class ProductAssociatedPartSQL
        {
            private Sys.Database _database;

            public ProductAssociatedPartSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public DataView GetAll_For_ProductID(int ProductID)
            {
                _database.ClearParameter();
                _database.AddParameter("@ProductID", ProductID, true);
                var dt = _database.ExecuteSP_DataTable("ProductAssociatedPart_GetAll_For_ProductID");
                dt.TableName = Sys.Enums.TableNames.tblProductAssociatedPart.ToString();
                return new DataView(dt);
            }

            public void Insert(ProductAssociatedPart oProductAssociatedPart)
            {
                _database.ClearParameter();
                AddProductAssociatedPartParametersToCommand(ref oProductAssociatedPart);
                _database.ExecuteSP_NO_Return("ProductAssociatedPart_Insert");
            }

            public void Delete(int ProductID)
            {
                _database.ClearParameter();
                _database.AddParameter("@ProductID", ProductID, true);
                _database.ExecuteSP_NO_Return("ProductAssociatedPart_Delete");
            }

            private void AddProductAssociatedPartParametersToCommand(ref ProductAssociatedPart oProductAssociatedPart)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@ProductID", oProductAssociatedPart.ProductID, true);
                    withBlock.AddParameter("@PartID", oProductAssociatedPart.PartID, true);
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Spares/Product/ProductAssociatedPart/ProductAssociatedPartSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace ProductAssociatedParts

        Public Class ProductAssociatedPartSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"

            Public Function [GetAll_For_ProductID](ByVal ProductID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@ProductID", ProductID, True)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("ProductAssociatedPart_GetAll_For_ProductID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblProductAssociatedPart.ToString
                Return New DataView(dt)
            End Function

            Public Sub [Insert](ByVal oProductAssociatedPart As ProductAssociatedPart)
                _database.ClearParameter()
                AddProductAssociatedPartParametersToCommand(oProductAssociatedPart)

                _database.ExecuteSP_NO_Return("ProductAssociatedPart_Insert")

            End Sub

            Public Sub Delete(ByVal ProductID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@ProductID", ProductID, True)
                _database.ExecuteSP_NO_Return("ProductAssociatedPart_Delete")
            End Sub

            Private Sub AddProductAssociatedPartParametersToCommand(ByRef oProductAssociatedPart As ProductAssociatedPart)
                With _database
                    .AddParameter("@ProductID", oProductAssociatedPart.ProductID, True)
                    .AddParameter("@PartID", oProductAssociatedPart.PartID, True)
                End With
            End Sub

#End Region

        End Class

    End Namespace

End Namespace


